Library/Homebrew/cmd/install.rb

```diff
@@ -278,48 +278,26 @@ def install
     # formula was found, but there's a problem with its implementation).
     $stderr.puts e.backtrace if Homebrew::EnvConfig.developer?
     ofail e.message
-  rescue FormulaOrCaskUnavailableError => e
-    if e.name == "updog"
+  rescue FormulaOrCaskUnavailableError, Cask::CaskUnavailableError => e
+    # formula name or cask token
+    name = e.try(:name) || e.token
+
+    if name == "updog"
       ofail "What's updog?"
       return
     end
 
     opoo e
-    ohai "Searching for similarly named formulae..."
-    formulae_search_results = search_formulae(e.name)
-    case formulae_search_results.length
-    when 0
-      ofail "No similarly named formulae found."
-    when 1
-      puts "This similarly named formula was found:"
-      puts formulae_search_results
-      puts "To install it, run:\n  brew install #{formulae_search_results.first}"
-    else
-      puts "These similarly named formulae were found:"
-      puts Formatter.columns(formulae_search_results)
-      puts "To install one of them, run (for example):\n  brew install #{formulae_search_results.first}"
-    end
+    ohai "Searching for similarly named formulae and casks..."
 
-    if (reason = MissingFormula.reason(e.name))
-      $stderr.puts reason
-      return
-    end
+    # Don't treat formula/cask name as a regex
+    query = string_or_regex = name
+    if search_names(query, string_or_regex, args)
+      puts <<~EOL
 
-    # Do not search taps if the formula name is qualified
-    return if e.name.include?("/")
-
-    taps_search_results = search_taps(e.name)[:formulae]
-    case taps_search_results.length
-    when 0
-      ofail "No formulae found in taps."
-    when 1
-      puts "This formula was found in a tap:"
-      puts taps_search_results
-      puts "To install it, run:\n  brew install #{taps_search_results.first}"
-    else
-      puts "These formulae were found in taps:"
-      puts Formatter.columns(taps_search_results)
-      puts "To install one of them, run (for example):\n  brew install #{taps_search_results.first}"
+        To install one of them, run:
+          brew install 'package_name'
```

Comment:
Can we use the `taps_search_results` like before to make this less generic?

Reply:
Yeah, I was lazy and originally went for the direct solution. I'm honestly not sure how necessary it is because to get here they have to know enough to use `brew install` at least once already.

I'd be fine with adding it back in but I'm afraid that it'll end up being even more verbose, since we're in theory including twice as much output as before.

```
$ brew install -n virtuabox
Warning: No available formula with the name "virtuabox". Did you mean virtuoso or virtualpg?
==> Searching for similarly named formulae and casks...
==> Formulae
virtuoso                                                                        virtualpg

To install virtuoso, run:
  brew install virtuoso

==> Casks
virtualbox ✔                                                                    virtualgl

To install virtualbox, run:
  brew install --cask virtualbox
```

Reply:
Thanks ❤️